### DISCUSSION TITLE
Older binaries for PowerPC

### DISCUSSION THREAD

**Discussion by mdbergmann** [Clozure/ccl#396] [General]
Hi.

I still have my old PowerPC iBook that is doing house automation jobs just fine.
I was wondering if I could do some of those tasks with Common Lisp and CCL.
Are there any CCL PowerPC binaries downloadable for MacOSX 10.4?

**Comment by xrme** [comment]
Please see https://ccl.clozure.com/download-1.7.html, which has a link for Darwin/ppc, but it needs 10.5.

https://trac.clozure.com/ccl/wiki/ReleaseNotes/1.4 has instructions on how to download a copy of CCL 1.4, which is the last release that supported Mac OS X 10.4. This is a link to the 1.4 release directory: https://ccl.clozure.com/ftp/pub/release/1.4/

**Reply by mdbergmann** [reply]
Thanks, I'll give this 1.4 release a try.

**Reply by mdbergmann** [reply]
Hello.

The release directories don't seem to be available anymore. I.e: https://ccl.clozure.com/ftp/pub/release/1.4/ I get 404 (Not Found) error.

**Reply by xrme** [reply]
I fixed the link. https://ccl.clozure.com/ftp/pub/release/1.4/ should work now.

**Reply by mdbergmann** [reply]
Thank you.

**Comment by barracuda156** [comment]
@mdbergmann Thank you for pointing to this topic. I will try implementing PPC back into Macports, and then hopefully it will be easily installable.

@xrme Do you know if somewhat later versions can be built for PPC using boot image from 1.6? I see 1.7 existed for Linux PPC, but not for MacOS.

**Reply by xrme** [reply]
CCL 1.7 works on PowerPC Macs running Mac OS X 10.5. https://ccl.clozure.com/download-1.7.html

If you're on Mac OS X 10.4, then CCL 1.4 is the latest version that runs on that OS. Later CCL versions need operating system features that aren't in Mac OS X 10.4.

Building a new version of CCL is unfortunately more complicated that just compiling newer sources with an older binary, so no, in general, you can't use, e.g., a 1.6 image to build 1.7 sources.

**Reply by barracuda156** [reply]
@xrme Hmm, I only find Linux PPC: https://ccl.clozure.com/ftp/pub/release/1.7/
Do I miss something? If Darwin PPC version exists, we certainly want it in Macports.

**Reply by xrme** [reply]
I'm sorry, I made a mistake about 1.7. It's 1.6 that runs on PowerPC Macs running Mac OS X 10.5.

**Reply by barracuda156** [reply]
@xrme It may be possible though to restore what was removed here: https://trac.clozure.com/ccl/browser/branches/ppc-purge/source
And then build new boot image for Darwin PPC from source using ccl 1.6 as a compiler.
Perhaps.

**Reply by xrme** [reply]
That ppc-purge branch was undone. I thought I wanted to get rid of the PPC-related sources, but Gary Byers said it would be better to leave them intact.

**Reply by barracuda156** [reply]
@xrme Cool. So what could be needed to restore Darwin PPC into the current version? Build new bootimages? Can that be done using `ccl` 1.6 (the last we have on PPC atm)?

**Reply by barracuda156** [reply]
@xrme I vaguely recall someone saying that CCL development history was moved to Github, however there are no tags for old version. I confuse something?

**Comment by xrme** [comment]
The oldest thing on GitHub is the 1.2 branch, which is pretty much as far as things went back on the prior Subversion repository.

For earlier stuff, all I have is in https://ccl.clozure.com/ftp/pub/

**Reply by barracuda156** [reply]
@xrme Oh, I just need history from 1.6 upwards (to try restoring Darwin PPC support – no concern for Tiger, 10.5/10.6 is good enough).
